printing/printer.ml

```diff
@@ -34,6 +34,49 @@ let delayed_emacs_cmd s =
 let get_current_context () =
   Pfedit.get_current_context ()
 
+let enable_unfocussed_goal_printing = ref false
+let enable_goal_tags_printing = ref false
+let enable_goal_names_printing = ref false
+
+let should_tag() = !enable_goal_tags_printing
+let should_unfoc() = !enable_unfocussed_goal_printing
+let should_gname() = !enable_unfocussed_goal_printing
+
+
+let _ =
+  let open Goptions in
+  declare_bool_option
+    { optsync  = false;
+      optdepr  = false;
+      optname  = "printing of unfocussed goal";
+      optkey   = ["Printing";"Unfocussed"];
+      optread  = (fun () -> !enable_unfocussed_goal_printing);
+      optwrite = (fun b -> enable_unfocussed_goal_printing:=b) }
+
+(* This is set on by proofgeneral proof-tree mode. But may be used for
+   other purposes *)
+let _ =
+  let open Goptions in
+  declare_bool_option
+    { optsync  = false;
+      optdepr  = false;
+      optname  = "printing of goal tags";
+      optkey   = ["Printing";"Goal Tags"];
```

Comment:
At first view, the parser will not be able to parse the space. You should do ["Printing";"Goal";"Tags"].

(Unless the space is intended, so that the option can only be used over a dedicated communication channel which accepts spaces in the name of the option?)